pkgs/development/python-modules/jenkinsapi/default.nix

```diff
@@ -29,6 +34,7 @@ buildPythonPackage rec {
   propagatedBuildInputs = [
     pytz
     requests
+    setuptools
```

Comment:
`setuptools` seems to be used by `jenkinsapi/utils/jenkins_launcher.py`. It seemed safer to put it in here but I'm not 100% sure if it's needed by downstream packages.

Reply:
```
$ rg setuptools -L results/
results/python311Packages.jenkinsapi/nix-support/propagated-build-inputs
1:/nix/store/cssc8bgd3bgg50zcsakzzbxb1cfp6f7q-python3.11-pytz-2024.1 /nix/store/lwz9pmmfsj76vgwbcf6k0k2h1wp4ma4h-python3.11-requests-2.31.0 /nix/store/4718wmk03wr3554kmf09vy80vkdjvq56-python3.11-setuptools-69.1.1 /nix/store/9z56ljqk4ahpx4fj01d82yr0dra6kyvl-python3.11-six-1.16.0 /nix/store/gd3shnza1i50zn8zs04fa729ribr88m9-python3-3.11.8 

results/python311Packages.jenkinsapi/lib/python3.11/site-packages/jenkinsapi-0.3.13.dist-info/METADATA
111:Egg-files for this project are hosted on PyPi. Most Python users should be able to use pip or setuptools to automatically install this project.

results/python312Packages.jenkinsapi/nix-support/propagated-build-inputs
1:/nix/store/lz8hmz046v1pqxl52wc7jqv85kzxdklm-python3.12-pytz-2024.1 /nix/store/jjh858xrjalznl2rl07fabx4xbghyr63-python3.12-requests-2.31.0 /nix/store/2bs5wmcfingkcwcm2ckd6n36hbacxjz0-python3.12-setuptools-69.1.1 /nix/store/isqpc21h7rk1hx255kwxdj463dzahy80-python3.12-six-1.16.0 /nix/store/fmwqa8nvva4sh18bqayzrilrzxq9fm0f-python3-3.12.2 

results/python312Packages.jenkinsapi/lib/python3.12/site-packages/jenkinsapi-0.3.13.dist-info/METADATA
111:Egg-files for this project are hosted on PyPi. Most Python users should be able to use pip or setuptools to automatically install this project.
```

i can find no import in the resulting builds

Reply:
Good idea to check that way! I've moved `setuptools` to `nativeCheckInputs`.

Reply:
Actually, you need to search for `pkg_resources` which comes from `setuptools`. I think the `__init__.py` occurrence is ok because it's under a `try` and appears to be for development, but the launcher I'm not entirely sure about.

```
❯ rg -l pkg_resources result/
result/lib/python3.11/site-packages/jenkinsapi/utils/jenkins_launcher.py
result/lib/python3.11/site-packages/jenkinsapi/__init__.py
```

Reply:
I added `setuptools` back to `propagatedBuildInputs`.

Reply:
I also added more to `pythonImportsCheck` to exercise the modules that rely on this.

Reply:
of course, i forgot about pkg_resources! I have a stale pr (#257527) which in effect primarily deals with that